airflow/jobs/scheduler_job.py

```diff
@@ -1408,6 +1416,40 @@ def _send_sla_callbacks_to_processor(self, dag: DAG) -> None:
         )
         self.executor.send_callback(request)
 
+    @provide_session
+    def _fail_tasks_stuck_in_queued(self, session: Session = NEW_SESSION) -> None:
```

Comment:
IMO this is a general problem that applies to both kubernetes & celery executors. The relevant k8s exec setting is [worker_pods_queued_check_interval](https://airflow.apache.org/docs/apache-airflow/stable/configurations-ref.html#worker-pods-queued-check-interval) -- I think this can also be handled in the scheduler. I also think this can probably replace [task-adoption-timeout](https://airflow.apache.org/docs/apache-airflow/stable/configurations-ref.html#task-adoption-timeout)

Reply:
worker_pods_queued_check_interval is similar, but different in that it won't automatically just reset the TI. It first checks to see if the pod exists.

worker_pods_pending_timeout is essentially this same process though. It should probably be deprecated as well (though, not sure how config handles many -> one).